### ISSUE TITLE
Full-text search

### ISSUE THREAD

**Issue by srid** [haskellfoundation/error-message-index#349]
For example, when I type `Character literals may not be empty` in the search box, I'd expect to see [GHC-11861](https://errors.haskell.org/messages/GHC-11861/index.html), but currently, it returns 0 results presumably because only the title is searched.

cf. https://stork-search.net/ | https://github.com/EmaApps/emanote/pull/327

**Comment by david-christiansen** [comment]
I agree that this would be nice! Stork-search looks really great.

I'm a bit reluctant to add a build-time dependency on the whole Rust toolchain for this site, though. I think it's important to keep things low-tech and very easy to maintain, both to make contributions by new users easier and to help me be efficient with my time use. If the instructions include "download this whole other compiler and language" or "install nix" or "use guix", then I think the barriers become too high.

Do you have any thoughts on how to reduce the maintenance and build-time implications of something like that?

**Comment by srid** [comment]
You are right. Doing something like this without Nix can be quite complex. 

Alternatively, you could parse the Markdown using pandoc and extract just the GHC error message, storing them as JSON that is then used by the JavaScript code.

But if search engines can index this site well enough such that it appears when people search for the specific error message, we probably wouldn't even need this feature.

**Comment by david-christiansen** [comment]
My hope is that Google and friends will do a good job indexing the site.

**Comment by david-christiansen** [comment]
On further reflection, I think that a setup where there was full-text search only of the GHC message, and not the documentation and explanations as well, could be quite confusing and unexpected.
